ms2/src/org/labkey/ms2/proteinCoverageMap.jsp

```diff
@@ -15,34 +15,160 @@
  * limitations under the License.
  */
 %>
+<%@ page import="org.apache.commons.lang3.StringUtils" %>
+<%@ page import="org.labkey.api.protein.ProteinFeature" %>
 <%@ page import="org.labkey.api.view.HttpView" %>
 <%@ page import="org.labkey.api.view.JspView" %>
-<%@ page import="org.labkey.ms2.MS2Controller" %>
 <%@ page import="org.labkey.api.view.template.ClientDependencies" %>
+<%@ page import="org.labkey.api.visualization.ColorGradient" %>
+<%@ page import="org.labkey.ms2.MS2Controller" %>
+<%@ page import="java.awt.*" %>
+<%@ page import="java.util.ArrayList" %>
+<%@ page import="java.util.HashMap" %>
+<%@ page import="java.util.List" %>
+<%@ page import="java.util.Map" %>
 <%@ page import="java.util.Set" %>
-<%@ page import="java.util.stream.Collectors" %>
 <%@ page import="java.util.TreeSet" %>
-<%@ page import="org.apache.commons.lang3.StringUtils" %>
-<%@ page import="org.labkey.api.protein.ProteinFeature" %>
+<%@ page import="java.util.stream.Collectors" %>
 <%@ page extends="org.labkey.api.jsp.JspBase" %>
 <%
     MS2Controller.ProteinViewBean bean = ((JspView<MS2Controller.ProteinViewBean>)HttpView.currentView()).getModelBean();
+    var currentURL = getActionURL();
+    var viewByParam = currentURL.getParameter("viewBy");
+    var isIntensityView = viewByParam == null || viewByParam.equalsIgnoreCase("intensity");
+    var isConfidenceView = viewByParam != null && viewByParam.equalsIgnoreCase("confidenceScore");

```

Comment:
We either are showing intensity of confidence, right? What happens if you have a viewBy parameter that doesn't match either?

Reply:
the old sequence coverage map (the current functionality prior to this) is displayed which also happens in the case when viewBy parameter is not present.